tests/external_repositories/CMakeLists.txt

```diff
@@ -0,0 +1,30 @@
+cmake_minimum_required(VERSION 3.0)
+project(external-repos)
+
+foreach(project Foo Bar)
+  set(build_dir "${CMAKE_BINARY_DIR}/${project}")
+  execute_process(
+    COMMAND "${CMAKE_COMMAND}"
+    "-H${CMAKE_SOURCE_DIR}/${project}"
+      "-B${build_dir}"
+      "-G${CMAKE_GENERATOR}"
+    RESULT_VARIABLE ${project}_result
+  )
+  if(NOT${project}_result EQUAL 0)
```

Comment:
This will do not what you're expecting. It will check value of variable with name `NOTFoo_result` instead of checking variable with name `Foo_result` and doing negation.